File: app/src/main/java/com/jtrofe/cheesebots/physics/objects/Particle.java
package com.jtrofe.cheesebots.physics.objects;

import android.graphics.Canvas;
import android.graphics.Paint;

import com.jtrofe.cheesebots.game.SpriteHandler;
import com.jtrofe.cheesebots.physics.Vec;

/**
 * A small particle for effects. Just has a color and a short lifespan
 */
public class Particle extends GameObject {

    public int LifeSpan;

    private int mColor;


    public Particle(Vec position, Vec velocity, int color, int lifeSpan){
        super(position);

        Type = GameObject.TYPE_PARTICLE;
        mLinearVelocity = velocity.copy();

        mFriction = 0.97;

        mColor = color;

        LifeSpan = lifeSpan;


        this.calculateMoment();
    }

    @Override
    public void Draw(Canvas canvas){
        SpriteHandler.PAINT.setStyle(Paint.Style.FILL);
        SpriteHandler.PAINT.setColor(mColor);

        canvas.drawCircle(mPosition.xf(), mPosition.yf(), (float) GameObject.PARTICLE_RADIUS, SpriteHandler.PAINT);
    }
}

File: app/src/main/java/com/jtrofe/cheesebots/physics/objects/Particle.kt
package com.jtrofe.cheesebots.physics.objects

import android.graphics.Canvas
import android.graphics.Paint
import com.jtrofe.cheesebots.physics.Vec

/**
 * A small particle for effects. Just has a color and a short lifespan
 */
public class Particle(position:Vec, velocity:Vec, color:Int, public var LifeSpan:Int):GameObject(position){

    var paint:Paint = Paint()

    init{
        Type = GameObject.TYPE_PARTICLE

        mLinearVelocity = velocity.copy()

        mFriction = 0.97

        paint.setColor(color)

        this.calculateMoment()
    }


    override fun Draw(canvas:Canvas){
        canvas.drawCircle(mPosition.xf, mPosition.yf, GameObject.PARTICLE_RADIUS.toFloat(), paint)
    }


}